tests/testsuite.src/run_misc.at

```diff
@@ -14570,3 +14570,32 @@ TST-DECIMAL IS < ZERO-DECIMAL
 
 
 AT_CLEANUP
+
+AT_SETUP([Decimal constants working after sub-program call])
+AT_KEYWORDS([runmisc])
+
+# this used to cause a SIGSEGV
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01  X  PIC 9(2).
+       PROCEDURE DIVISION.
+           CALL "prog2"
+           IF X + 42 <> 0
```

Comment:
to make that portable we should give it a `VALUE` and to ensure that also future versions of the compiler don't optimize the constant away too easily (trunk already uses integer arithmetic in most possible places) we need to make that more complex. I suggest the following (that's not ideal as it won't work with all `-std`, the only one that would work is replacing that function by `ACCEPT FROM DATE` to a 6 digit variable and use that)

```cobol
       01  X  PIC 9(2) VALUE 0.
       01  Y  PIC 9v9  VALUE 0.1.
       PROCEDURE DIVISION.
       MAIN.
      *    ensure that cobc cannot optimize the expression away
           IF FUNCTION CURRENT-DATE() = 0
              ADD 1 TO Y.
           CALL "prog2"
           IF X + Y + 42.1 <> 0
```